nodejs/eks/nodegroup.ts

```diff
@@ -652,19 +653,54 @@ export function createNodeGroup(
     return createNodeGroupInternal(name, args, pulumi.output(core), parent, provider);
 }
 
+export function resolveOrGetInstanceProfile(
+    name: string,
+    args: Omit<NodeGroupOptions, "cluster">,
+    c: pulumi.UnwrappedObject<CoreData>,
+    parent: pulumi.ComponentResource,
+    provider?: pulumi.ProviderResource,
+): aws.iam.InstanceProfile {
+    if (
+        !args.instanceProfile &&
+        !args.instanceProfileName &&
+        !c.nodeGroupOptions.instanceProfile &&
+        !c.nodeGroupOptions.instanceProfileName
+    ) {
+        throw new pulumi.ResourceError(
+            `an instanceProfile or instanceProfileName is required`,
+            parent,
+        );
+    }
+    if (
+        (args.instanceProfile || c.nodeGroupOptions.instanceProfile) &&
+        (args.instanceProfileName || c.nodeGroupOptions.instanceProfileName)
```

Comment:
You could simplify this with null coalescing:
```suggestion
        (args.instanceProfile?.instanceProfile && c.nodeGroupOptions?.instanceProfileName)
```

Reply:
Mhm wait, my suggestion is not correct.

Aren't we susceptible to null pointer issues though? Is `c.nodeGroupOptions` guaranteed to be set?

Reply:
Just checked, we're good. `c.nodeGroupOptions` is set even when we're not creating the default node group. All good!